public/src/Components/Heroes.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { JsxElement } from 'typescript';
+import {getAPIResource, getAllHeroes, getHeroeByID, getAllComicsOfHero, getSearchedfHeroes} from '../api/api'
+import ComicsOfHero from './ComicsOfHero';
+import { NavLink } from 'react-router-dom';
+
+const queryString = require('query-string');
+
+
+class Heroes extends React.Component<any, any> {
+    constructor(props:any) {
+        super(props) 
+        this.state = {
+            heroes: [],
+            searchInput: "",
+        } 
+        this.handleChange = this.handleChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleChange(e: any){
+        this.setState({searchInput: e.target.value});
+    }
+    
+    handleSubmit(e: any){
+        e.preventDefault();
+    console.log(this.props);
+    let {searchInput} = this.state;
+    this.props.history.push(`?query=${searchInput}`);
+    }
+
+    componentDidMount() {
+        getAllHeroes().then((res) => {
+            this.setState({heroes: res.data.data.results})
+        })
+        
+    }
+
+    componentDidUpdate(prevProps: any){
+        if (this.props.location !== prevProps.location) {
+            console.log(this.props.location.search);
+            const wantedHeroName= queryString.parse(this.props.location.search).query;
```

Comment:
bad variable name

Reply:
If I replace it with searchedHeroName, will it be good?